app/scripts/controllers/transactions/tx-state-manager.js

```diff
@@ -521,7 +521,7 @@ export default class TransactionStateManager extends EventEmitter {
 
     const txMeta = this.getTransaction(txId);
     txMeta.err = {
-      message: error.toString(),
+      message: typeof error === 'object' ? error.message : error.toString(),
```

Comment:
Small may be `error.message?.toString() || error.toString()`

Reply:
I haven't seen cases where we need to stringify the error messages from the ledger library having tested most of the common errors. 

The idea to convert to string I believe is to account for any, and all, error formats from different providers as there is not a common error format across the board.

Reply:
Resolved in https://github.com/MetaMask/metamask-extension/pull/12176/commits/aa13c0f823fa1115d2cee6baa89c39251900665c.